persist/log.go

```diff
@@ -3,28 +3,65 @@ package persist
 import (
 	"log"
 	"os"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
+// logFileWrapper wraps a log file to perform sanity checks on the Write and
+// Close functions. An error will be returned if Close is called multiple times
+// or if Write is called after Close has been called.
+type logFileWrapper struct {
+	closed bool // Flag is set after Close is called.
+	file   *os.File
+}
```

Comment:
right right, that makes sense. I guess I was pretty tired as well.

it seems like this type only exists to enable a specific sanity check though. Didn't you originally implement this to prevent a race condition? A race condition should be just as visible as a panic.

I'm actually starting to question whether we really need to close the log file. What does that do for us? It's only relevant during testing, and we certainly don't have to worry about opening too many file handles or something. Our primary focus should be shutting down all goroutines, because those can have cross-test side effects.
More importantly, using an *os.File in the persist logger makes it harder to mock. One of the advantages of a standard `log.Logger` is that you can log to an in-memory Writer, like a `bytes.Buffer`.